File: src/main/java/dev/canon/cue/module/modules/function/NoRotate.java
package dev.canon.cue.module.modules.function;

import dev.canon.cue.event.events.EventStage;
import dev.canon.cue.event.events.client.PacketEvent;
import dev.canon.cue.event.events.player.UpdateWalkingPlayerEvent;
import dev.canon.cue.module.Category;
import dev.canon.cue.module.Module;
import dev.canon.cue.module.ModuleInfo;
import dev.canon.cue.settings.ModeSetting;
import dev.canon.cue.module.Category;
import dev.canon.cue.module.Module;
import dev.canon.cue.module.ModuleInfo;
import net.minecraft.network.play.client.CPacketPlayer;
import net.minecraft.network.play.server.SPacketPlayerPosLook;
import net.minecraftforge.fml.common.eventhandler.Event;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

import java.util.concurrent.ThreadLocalRandom;

@ModuleInfo(name = "NoRotate", descriptions = "Dangerous to use might desync you.", category = Category.FUNCTION)
public class NoRotate extends Module {
    private boolean cancelPackets = true;
    private boolean timerReset = false;

    @Override
    public void onUpdate() {
        if (this.timerReset && !this.cancelPackets) {
            this.cancelPackets = true;
            this.timerReset = false;
        }
    }

    @SubscribeEvent
    public void onPacketReceive(PacketEvent.Receive event) {
        if (fullNullCheck()) return;
        if (event.getStage() == 0 && this.cancelPackets && event.getPacket() instanceof SPacketPlayerPosLook) {
            SPacketPlayerPosLook packet = event.getPacket();
            packet.yaw = NoRotate.mc.player.rotationYaw;
            packet.pitch = NoRotate.mc.player.rotationPitch;
        }
    }

}

File: src/main/java/cn/origin/cube/module/modules/function/NoRotate.kt
package cn.origin.cube.module.modules.function

import cn.origin.cube.core.events.client.PacketEvent
import cn.origin.cube.core.module.Category
import cn.origin.cube.core.module.Module
import cn.origin.cube.core.module.interfaces.ModuleInfo
import net.minecraft.network.Packet
import net.minecraft.network.play.server.SPacketPlayerPosLook
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

@ModuleInfo(name = "NoRotate", descriptions = "Dangerous to use might desync you.", category = Category.FUNCTION)
class NoRotate: Module() {

    private var cancelPackets = true
    private var timerReset = false

    override fun onUpdate() {
        if (timerReset && !cancelPackets) {
            cancelPackets = true
            timerReset = false
        }
    }

    @SubscribeEvent
    fun onPacketReceive(event: PacketEvent.Receive) {
        if (fullNullCheck()) return
        if (event.stage == 0 && cancelPackets && event.getPacket<Packet<*>>() is SPacketPlayerPosLook) {
            val packet = event.getPacket<SPacketPlayerPosLook>()
            packet.yaw = mc.player.rotationYaw
            packet.pitch = mc.player.rotationPitch
        }
    }
}